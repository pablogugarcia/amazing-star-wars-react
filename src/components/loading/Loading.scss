@import '../../themes/colors.scss';

.loading {
    width: 200px;
    height: 200px;
    clip-path: circle(17.1% at 50% 50%);
    animation: circle 5s infinite;
}

.loading-container {
    display: flex;
    justify-content: center;
    flex-direction: column;
    align-items: center;
}

.loading-label {
    color: $primary;
}

@keyframes circle {
    0% {
        clip-path: circle(17.1% at 50% 50%);
    }
    50% {
        clip-path: circle(50% at 50% 50%);
    }
    100% {
        clip-path: circle(17.1% at 50% 50%);
    }
}
