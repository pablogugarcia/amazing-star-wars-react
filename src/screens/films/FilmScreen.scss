@import '../../themes/colors.scss';

.film-screen-container {
    display: grid;
    grid-template-columns: repeat(3, 1fr);
    grid-gap: 10px;
    grid-auto-rows: minmax(100px, auto);

    .film-card-content {
        display: flex;
        justify-content: space-around;
        .film-label {
            font-weight: 700;
        }
    }
}
