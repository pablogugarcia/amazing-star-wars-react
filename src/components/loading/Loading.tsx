import React from 'react'
import LoadingGIF from '../../assets/loading.gif'
import './Loading.scss'

const Loading = () => (
    <div className="loading-container">
        <img src={LoadingGIF} alt="LOADING" className="loading" />
        <span className="loading-label">Cargando...</span>
    </div>
)

export default Loading
