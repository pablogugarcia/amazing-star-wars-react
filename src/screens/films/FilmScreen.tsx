import React, { useEffect, useState, useContext } from 'react'
import './FilmScreen.scss'
import { Card, CardContent } from '@material-ui/core'
import { Link } from 'react-router-dom'
import { Film } from '../../models/api/Api'
import MovieCard from '../../components/movie-card/MovieCard'
import Loading from '../../components/loading/Loading'
import { store } from '../../store/store'
import { getFilms } from '../../store/api/api-actions'

const FilmScreen = (): JSX.Element => {
  const [films, setFilms] = useState<(Film & { isSelected?: boolean })[]>([])
  const globalState = useContext(store)
  const { dispatch, state } = globalState

  useEffect(() => {
    const id = setTimeout(() => {
      if (state.films.length === 0) {
        getFilms(dispatch)
      }
    }, 3000)

    return () => clearTimeout(id)
  }, [])

  useEffect(() => { 
    setFilms(state.films)
  }, [state.films])

  const handleCardClick = (title: string) => () => {
    setFilms((prevState) =>
      prevState.map((film) => {
        if (film.title === title) {
          const newFilm = { ...film }
          newFilm.isSelected = !film.isSelected
          return newFilm
        }
        return film
      })
    )
  }

  if (state.films.length === 0) {
    return <Loading />
  }

  return (
    <div className="film-screen-container">
      {films.map((film) => (
        <Card
          key={film.title}
          className="film-card"
          style={{
            gridColumnStart: film.isSelected ? 'span 3' : 'unset',
          }}
          onClick={handleCardClick(film.title)}
        >
          <CardContent className="film-card-content">
            <div>
              <h3>
                <span className="film-label">Movie name: </span>
                {film.title}
              </h3>
              <p>
                <span className="film-label">Episode number : </span>
                {film.episode_id}
              </p>
              <p>
                <span className="film-label">Director: </span>
                {film.director}
              </p>
              <p>
                <Link to="/People">People</Link>
              </p>
            </div>

            {film.isSelected && <MovieCard text={film.opening_crawl} />}
          </CardContent>
        </Card>
      ))}
    </div>
  )
}

export default FilmScreen
