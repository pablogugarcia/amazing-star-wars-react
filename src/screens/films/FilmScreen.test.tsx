/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable no-undef */
import React from 'react'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import FilmScreen from './FilmScreen'
import Loading from '../../components/loading/Loading'

configure({ adapter: new Adapter() })

describe('<FilmScreen/>', () => {
  const container = shallow(<FilmScreen />)

  it('should match the snapshot', () => {
    expect(container.html()).toMatchSnapshot()
  })

  it('should have loading in flims length 0', () => {
    expect(container.contains(<Loading />)).toEqual(true)
  })

})
