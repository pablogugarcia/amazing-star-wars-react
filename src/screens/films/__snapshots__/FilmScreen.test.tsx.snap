// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilmScreen/> should match the snapshot 1`] = `"<div class=\\"loading-container\\"><img src=\\"loading.gif\\" alt=\\"LOADING\\" class=\\"loading\\"/><span class=\\"loading-label\\">Loading...</span></div>"`;
